### ISSUE TITLE
Browser Extension: Hint for already added pages

### ISSUE THREAD

**Issue by starise** [omnivore-app/omnivore#2747]
It'd be very useful if the extension would signal that a web page has already been added to Omnivore. As an example, this is the behavior of Raindrop.io extension (I'm using Firefox on Linux).

![Firefox Raindrop icon vs Omnivore icon](https://i.imgur.com/uvQaK7i.png)

When I visit a page that's already been added to Raindrop, it makes me understand that using a blue checkmark. When I click the icon It opens the bookmark for editing. Right now Omnivore just re-add the page resetting the "added" date. I think it's better to open the bookmark page to let users add notes and tags to the existing articles as suggested in #2224.

**Comment by kquinsland** [comment]
I'd like to see this as well, but only if implemented in a "plugin pulls the URLs" model and not a "plugin sends each new tab URL to `$remote` to see if there's a match".

**Comment by jacksonh** [comment]
Yeah, syncing all your local URLs would be nicer than the way raindrop does it (sending your full browsing history to the server). I'd have to think about the best way to fetch/update that list though.

**Comment by kquinsland** [comment]
> Yeah, syncing all your local URLs would be nicer than the way raindrop does it (sending your full browsing history to the server). 

[This](https://www.troyhunt.com/understanding-have-i-been-pwneds-use-of-sha-1-and-k-anonymity/) is a technique that's meant to preserve privacy but scale well.  Granted, it needs to scale to a "very many to one" pattern and not a "few instances of extension pulling one per person" pattern though.

Hashes are easy/cheap-ish to calculate and the constant size (~20 bytes) makes sizing/storing the lists easy to plan for compared to variable URL length.

> I'd have to think about the best way to fetch/update that list though.

Hash immediately when saved through browser extension and periodically fetch the hashes from server and merge or websocket push down to the extension - if that's permitted by the browsers.

/just ideas

**Comment by jacksonh** [comment]
I don't think we'd want a websocket open to every installed version of the extension. Probably neither users or my checkbook would like that :), but we can probably find some times to strategically call our sync API to get new URLs.

**Comment by jacksonh** [comment]
Using the hashes is a great idea, we can probably build this API somewhat ontop of our syncUpdates API.

**Comment by kquinsland** [comment]
> I don't think we'd want a websocket open to every installed version of the extension. Probably neither users or my checkbook would like that :), but we can probably find some times to strategically call our sync API to get new URLs.

Fair enough! Having the refresh interval adjustable from something like 10 to 240 min; default 15 is probably fine for most people and/or make WSS a premium feature for the people that need near-real-time. Thinking through how I use apps like omnivore/pocket...etc, it's _super_ unlikely that I'll try adding the same link more than once in a period that can probably be measured in weeks. Usually it's some re-post on hacker news that I go to save and then I get the "you've already added this; we'll move it to the top of the queue" notification.

**Comment by Podginator** [comment]
Closing due to age, and the self-hosted nature of Omnivore. Done as a batch action. If this issue still exists, please reopen it.
